### DISCUSSION TITLE
Dose polygon-edge support smart contract?

### DISCUSSION THREAD

**Discussion by JSHan94** [0xPolygon/polygon-edge#411] [Q&A] [answered]
I wonder if i can deploy a smart contract on my own blockchain developed with polygon-edge.

I want to mint a token in polygon-edge based blockchain and swap in the erc20 token in Ethereum.

It's hard to find an explanation for that part even if i search the official documentation.

Please help me!

Our requirements is as follows.

1. Support solidity based smart contract
2. Support block explorer and compatibility with metamask
3. Support easy token bridging functionality

Is polygon-edge suitable these requirements?

**Comment by Kourin1996** [comment, marked as answer]
@JSHan94 Hello, Thank you for your question. You can deploy contract in the same manner as Ethereum. You will need to set bytecode to `input` field and keep `to` field blank in transaction, and send request to `eth_sendRawTransaction` in JSON-RPC. https://polygon-edge.netlify.app/docs/get-started/json-rpc-commands#eth_sendrawtransaction

If you're using Remix, you can deploy contract by Injected Web3 via Metamask.

![Screenshot 2022-03-01 at 12 16 30](https://user-images.githubusercontent.com/19720977/156150824-0df901cb-96cd-4412-8e77-d4f5a1331242.png)

**Reply by JSHan94** [reply]
Thank you so much.

**Reply by tienduongnguyen** [reply]
Thank you, but how can I verify the contract?

**Reply by Kourin1996** [reply]
@tienduongnguyen Hello. Could you tell me what `verify` means? You can call the method of the deployed contract via JSON-RPC.
